### PR TITLE
(SIMP-4663) Pin default tmp_service to system state

### PR DESCRIPTION
**Pull request by trevor-vaughan**
When running in the wild, it was discovered that it is unlikely that
switching over to the tmp.mounts service will actually be able to
succeed.

The module was changed to set simp::mountpoints::tmp::tmp_service to
`false` by default and `true` if the existing `/tmp` mountpoint has a
filesystem type of `tmpfs`.

SIMP-4663 #close

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/58222741

File: spec/classes/mountpoints/tmp_spec.rb [hunk 1]
```diff
@@ -80,18 +87,12 @@
...
-What=tmpfs
-Where=/tmp
-Type=tmpfs
-Options=mode=1777,nodev,noexec,nosuid
-                  EOM
-                )
-            }
+            it { is_expected.to contain_mount('/tmp').with({
+              :options => 'data=ordered,nodev,noexec,nosuid,relatime,rw,seclabel',
+              :device  => '/dev/sda3'
```
Comment (sharkbruhaha) [inline spec/classes/mountpoints/tmp_spec.rb]: Do we not need to be checking `mode` as well?

Comment (trevor-vaughan) [reply spec/classes/mountpoints/tmp_spec.rb]: @sharkbruhaha Not really. That isn't defined by variable logic and we would just then be re-checking something that's "just puppet".

This gets back to the maintainability of 100% code coverage.

**Comment by pillarsdotnet** [comment]
The module should probably issue an explanatory warning if the user explicitly sets ` simp::mountpoints::tmp::tmp_service` to `true` and Puppet is unable to remount without a reboot.

**Comment by trevor-vaughan** [comment]
@pillarsdotnet Unfortunately, there's no real way to know if that's going to happen. The last PR made it thorough because nothing was locking in `/tmp`. I could update the comments to that effect to at least provide some documentation around the issue but I don't think I can reliably hook in a `reboot_notify` without a lot of confusion.

**Comment by pillarsdotnet** [comment]
Doesn't puppet have some kind of "try/catch/except" mechanism?  (...googling...)  Hm... There's a [module](https://github.com/dalen/puppet-trycatch) for that.

> @pillarsdotnet Unfortunately, there's no real way to know if that's going to happen. The last PR made it thorough because nothing was locking in `/tmp`. I could update the comments to that effect to at least provide some documentation around the issue but I don't think I can reliably hook in a `reboot_notify` without a lot of confusion.

**Comment by trevor-vaughan** [comment]
@pillarsdotnet That's server side. The issue is that this is a client-side issue which would mean catalog mangling based on the execution of the service resource.

Technically, this could be done, but it's non-trival and enters the real of "things that make Henrik sad".

**Comment by pillarsdotnet** [comment]
Another way would be to write a custom fact that does the equivalent of `lsof /tmp`, and wrap the remount in a conditional like:
```
if ($facts.opentmpfiles > 0) {
  warn('You'll need to reboot to remount /tmp')
} else {
  do-the-remount-thingie()
}
```

**Comment by trevor-vaughan** [comment]
@pillarsdotnet That's valid. I think we should put that off for a bit though. I've had cases where `lsof` would take a LONG time to return and wouldn't want to burden facter with that.

If we can come up with a safer way of doing this, it might be a viable addition.

**Comment by pillarsdotnet** [comment]
There are more efficient ways than `lsof` to be sure.

**Comment by trevor-vaughan** [comment]
Added [SIMP-6510](https://simp-project.atlassian.net/browse/SIMP-6510) for follow on.

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/58246596

**Pull request merged**